Service/src/routes/portfolio.py

```diff
@@ -30,6 +30,46 @@ def post_portfolio():
    return jsonify(id=portfolioId, buyPower=buyPower)
 
 
+@portfolio_api.route('/api/portfolios', methods=['GET'])
+@auth.login_required
+@validator.validate_params(portfolio_get_schema.fields)
+def get_portfolios():
+   leagueId = request.args.get('leagueId')
+   if leagueId:
+      # validate that user belongs to the league
+      # fetch all portfolios that belong to league
+      return Response(status=405)
+   
+   else:
+      g.cursor.execute("SELECT id, buyPower, name, userId, leagueId FROM Portfolio " + 
+         "WHERE userId = %s", g.user['id'])
+
+      portfolios = g.cursor.fetchall()
+      
+   for portfolio in portfolios:
+      attach_portfolioItems(portfolio)
+      attach_portfolio_netWorth(portfolio)
+
+   return json.dumps(portfolios)
+   
+
+def attach_portfolioItems(portfolio):
+   items = []
+   g.cursor.execute("SELECT id, shareCount, avgCost, ticker FROM PortfolioItem WHERE portfolioId = %s", portfolio['id'])
+   items = g.cursor.fetchall()
+
+   portfolio['items'] = items
+
+
+def attach_portfolio_netWorth(portfolio):
+   netWorth = float(portfolio['buyPower'])
+   if len(portfolio['items']) > 0:
```

Comment:
Is this check necessary? Wouldn't it just skip the for loop if the length isn't greater than 0?

Reply:
yea I guess that's true huh